modules/exploits/solaris/local/libnspr_nspr_log_file_priv_esc.rb

```diff
@@ -0,0 +1,219 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+
+  include Msf::Post::File
+  include Msf::Post::Solaris::Priv
+  include Msf::Post::Solaris::System
+  include Msf::Post::Solaris::Kernel
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Solaris libnspr NSPR_LOG_FILE Privilege Escalation',
+      'Description'    => %q{
+        This module exploits an arbitrary file write vulnerability in the
+        Netscape Portable Runtime library (libnspr) on Solaris which allows
+        users to gain root privileges.
+
+        libnspr versions prior to 4.6.3 allow users to specify a log file with
+        the `NSPR_LOG_FILE` environment variable. The log file is created with
+        the privileges of the running process, resulting in privilege escalation
+        when used in combination with a SUID executable.
+
+        This module writes a shared object to the trusted library directory
+        `/usr/lib/secure` and runs the specified SUID binary with the shared
+        object loaded using the `LD_LIBRARY_PATH` environment variable.
+
+        This module has been tested successfully with libnspr version 4.5.1
+        on Solaris 10u1 (01/06) (x86) and Solaris 10u2 (06/06) (x86).
+      },
+      'References'     =>
+        [
+          ['BID', '20471'],
+          ['CVE', '2006-4842'],
+          ['EDB', '2543'],
+          ['EDB', '2569'],
+          ['EDB', '2641'],
+          ['URL', 'https://securitytracker.com/id/1017050'],
+          ['URL', 'https://securitytracker.com/id/1017051'],
+          ['URL', 'https://developer.mozilla.org/en-US/docs/Mozilla/Projects/NSPR'],
+          ['URL', 'http://web.archive.org/web/20061118024339/http://labs.idefense.com:80/intelligence/vulnerabilities/display.php?id=418'],
+          ['URL', 'http://web.archive.org/web/20061110164829/http://sunsolve.sun.com/search/document.do?assetkey=1-26-102658-1']
+        ],
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'iDefense',     # Discovery
```

Comment:
I like the simplified credits. I'm still deciding if I want to keep it short or write in more detail. I may adopt this in the future.